### ISSUE TITLE
Centering text in the middle/wrapping

### ISSUE THREAD

**Issue by xpresas** [Bodmer/TFT_eSPI#3075]
Only raise issues for problems with the library and/or provided examples. Post questions, comments and useful tips etc in the "Discussions" section.

To minimise effort to resolve issues the following should be provided as a minimum:
1. A description of the problem and the conditions that cause it to occur
2. IDE (e.g. Arduino or PlatformIO)
3. TFT_eSPI library version (try the latest, the problem may have been resolved!) from the Manage Libraries... menu
4. Board package version (e.g. 2.0.3) available from the Boards Manager... menu
5. Procesor, e.g RP2040, ESP32 S3 etc
6. TFT driver (e.g. ILI9341), a link to the vendors product web page is useful too.
7. Interface type (SPI or parallel)

Plus further information as appropriate to the problem:
1. TFT to processor connections used
2. A zip file containing your setup file (just drag and drop in message window - do not paste in long files!)
3. A zip file  containing a simple and complete example sketch that demonstrates the problem but needs no special hardware sensors or libraries.
4. Screen shot pictures showing the problem  (just drag and drop in message window)

The idea is to provide sufficient information so I can setup the exact same (or sufficiently similar) scenario to investigate and resolve the issue without having a tedious ping-pong of Q&A.

DO NOT paste code directly into the issue. To correctly format code put three ticks ( ` character on key next to "1" key) at the start and end of short pasted code segments to avoid format/markup anomolies. [See here:](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code)

Example output:

1. Simply Im writing a program that fetches direct messagess from discord bot api and I want to display them on the screen. Ive run into multiple problems which most of them I managed to make it work. (Playing gifs, uploading them to littlefs, swapping gifs and etc).
Tbh I havent checked ALL of the examples because some are outdated and theres so much of them I just thought that Im going to write here and I might get help.
The problem Im facing right now is pretty simple.  I pull a message from discords api. That message contains a handful of words. I cant seem to find a way to 

- wrap text
- display it in the middle of the screen
- consistenly stay at that location. (I tried to use https://github.com/Bodmer/TFT_eSPI/issues/558#issuecomment-593147012 this) and after the loop refreshes the text just moves bit by bit.
- A way to display emojis pulled from that text.

2. Arduino v1.
3. Latest
4. Latest esp32 2.0.14
5. ESP32-Wroom-32U (same old esp, just 38pin version)
6. ST7789_DRIVER (320x240 from aliexpress)
7. SPI

```
    tft.fillScreen(TFT_BLACK);
    tft.setTextColor(TFT_WHITE);
    tft.setTextSize(2);
    tft.setTextDatum(CC_DATUM);
   //this is weird, and I bet there is an easier way to manage this
    tft.drawString("Test1",tft.width() / 2,tft.height() / 2 - 15);
    tft.drawString("Test2",tft.width() / 2,tft.height() / 2 + 0);
    tft.drawString("Test3",tft.width() / 2,tft.height() / 2 + 15);
```
Basically if it would be possible to provide an example of how would I go about centering a text string while also wrapping it. I would appreciate it.

**Comment by Bodmer** [comment]
You will need to write code to split strings at an appropriate point and then plot each segment onto the screen. I do not have further example code to do this.
